components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -104,10 +107,57 @@ function Main() {
     }
   }
 
+  let inputFocused: boolean = false
+  let viewPortWithoutKeyboard = 0
+  let keyboardSize = 0
+
+  window.addEventListener('resize', () => {
+    if (!context.isMobile || !inputFocused || window.visualViewport == null) {
+      return
+    }
+    let viewPortWithKeyboard = window.visualViewport.height
+    if (headerElement == null || conversationContentElement == null ||
+      viewPortWithKeyboard === 0 || viewPortWithoutKeyboard === 0) {
+      return
+    }
+    if (keyboardSize === 0 ||
+        keyboardSize < viewPortWithoutKeyboard - viewPortWithKeyboard) {
+      keyboardSize = viewPortWithoutKeyboard - viewPortWithKeyboard
+    }
+    let mountPoint: HTMLElement | null = document.getElementById('mountPoint')
+    if (mountPoint !== undefined && mountPoint !== null) {
+      if (mountPoint.clientHeight >=
+          (headerElement.clientHeight + conversationContentElement.clientHeight)
+          * 2) {
+        let percent = viewPortWithKeyboard * 100 / viewPortWithoutKeyboard
+        mountPoint.style.height = `${percent}%`
+      } else if (keyboardSize > viewPortWithoutKeyboard - viewPortWithKeyboard) {
+        mountPoint.style.height = '100%'
+      }
+    }
+  })
+
+  const handleOnFocusInput = () => {
+    inputFocused = true
+    if (window.visualViewport != null) {
+      viewPortWithoutKeyboard = window.visualViewport.height
+    }
+  }
+
+  const handleOnBlurInput = () => {
+    let mountPoint: HTMLElement | null = document.getElementById('mountPoint')
+    if (mountPoint !== undefined && mountPoint !== null
```

Comment:
nit: merge these into a single check - we don't care about the difference between null/undefined here. Additionally, `getElementById` will never return undefined.

```suggestion
    if (mountPoint
```

we can merge even further and do

```suggestion
    if (mountPoint?.style.height !== '100%'
```